src/ChannelServer/World/Inventory.cs

```diff
@@ -232,12 +271,30 @@ public int AddSilent(Item item)
 		/// <return>Index of the item.</return>
 		public int Add(Item item, InventoryAddType addType)
 		{
-			var index = this.AddSilent(item);
+			Dictionary<int, int> indexes = this.AddSilent(item);
+
+			var lastItem = indexes.Last();
+			int lastIndex = lastItem.Key;
+
+			foreach (var itemObject in indexes)
+			{
+				int index = itemObject.Key;
+				int diff = itemObject.Value;
+				
+				if (index == lastIndex)
+				{
+					addType = InventoryAddType.PickUp;
```

Comment:
Hm... you're overriding the parameter here. If you `Add` something with a certain add type, it shouldn't use PickUp for the last, but whatever you put in.